addons/transitions/Transitions.gd

```diff
@@ -5,16 +5,28 @@ signal pre_transition
 const FadeScene = preload("res://addons/transitions/FadeScene.tscn")
 var _root:Viewport
 var _current_scene = null
+var scene_container:Node setget _set_scene_container
 
 enum FadeType {
 	Instant,
 	CrossFade,
 	Blend
 }
 
+func _set_scene_container(new_container:Node):
+	if new_container == null:
+		push_error("Can't change scene container to null scene!")
+	# Allow users to specify their own scene container node.
+	# This method also sets the current scene to be the last child of the new container.
+	scene_container = new_container
+	_current_scene = scene_container.get_child(scene_container.get_child_count() - 1)
```

Comment:
Can we extract this logic (get last child node and assume it's the current scene) into a private method (underscore-prefix)?

In one of my previous games, I added a CanvasModulate and/or Tween to some places, and it complicated this assumption. For now, it's OK to just extract it into a method.

Reply:
So a `_current_scene()` method that returns the scene container's last child? And replacing the `_current_scene` variable with calls to that method? Good idea.

Reply:
I've gone through and done this on my local copy, and nothing is obviously broken. My transitions still work.